src/app/api/auth/login/github/callback/route.ts

```diff
@@ -24,6 +24,22 @@ export const GET = async (request: NextRequest) => {
       },
     });
     const githubUser: GitHubUser = await githubUserResponse.json();
+    if(!githubUser.email) {
+      const githubEmailsResponse = await fetch("https://api.github.com/user/emails", {
+        headers: {
+          Authorization: `Bearer ${tokens.accessToken}`,
+        },
+      });
+      const githubEmails: { email: string; primary: boolean; verified: boolean; }[] = await githubEmailsResponse.json();
+      const verifiedEmail = githubEmails.find((email) => email.primary && email.verified);
+      if (!verifiedEmail) {
+        return new Response(null, {
+          status: 400,
+        });
+      }
+      githubUser.email = verifiedEmail.email;
```

Comment:
IMO, no need to throw 404 or anything for unverified mail, if verified then `emailVerified: true` or false.